### PR TITLE
Swap refactor

### LINKED ISSUE #28: Swap flow refactoring
A flow like:

1. amounts
2. invoice
3. refund file
4. address and QR code with button that waits for events

would make more sense because:

- the user gets the refund file before he sends any funds
- the screen on which the user sends onchain funds is also the one on which he waits for the transaction to confirm

Any opinions on that one? @ImmanuelSegol @peartobear

### PR DESCRIPTION
**Pull request by ImmanuelSegol**
Closes #28

### CODE REVIEW

File: src/views/swap/steps.js [hunk 1]
```diff
@@ -135,33 +140,22 @@ const StyledStepTwo = ({ classes, swapInfo, swapResponse }) => (
...
+          {' '}
+          {toWholeCoins(swapResponse.expectedAmount)} {swapInfo.base}{' '}
         </b>{' '}
-        to:
+        <br />
+        on <b>{getCurrencyName(swapInfo.base)}</b> <br />
+        blockchain address:
       </p>
-      <p className={classes.address} id="copy-address">
+      <p id="copy-address" className={classes.address}>
```
Comment (michael1011) [inline src/views/swap/steps.js]: I renamed the id of the element to copy to just `copy` and moved the `copyToClipBoard` method into the utils folder in the [reverse swap PR](https://github.com/BoltzExchange/boltz-frontend/pull/49/files#diff-13f3ac8aec59f01de2a366e701202512R81).

If it gets merged first keep that in mind when resolving the conflicts

File: src/views/swap/steps.js [hunk 2]
```diff
@@ -135,33 +140,22 @@ const StyledStepTwo = ({ classes, swapInfo, swapResponse }) => (
...
       </p>
-      {/* TODO: refactor how we copy */}
       <span className={classes.action} onClick={() => copyToClipBoard()}>
         Copy
       </span>
-      <p>
-        If the address does not work with your wallet:{' '}
-        <a
-          target={'_blank'}
-          href="https://litecoin-project.github.io/p2sh-convert/"
```
Comment (michael1011) [inline src/views/swap/steps.js]: @peartobear  why did you remove that from the mockup? We need it for wallet compatibility.

Comment (peartobear) [reply src/views/swap/steps.js]: I thought about putting it in the banner UI but I think we should just put it back here for now

Comment (michael1011) [reply src/views/swap/steps.js]: @peartobear I am still not exactly sure what kind of information we should show in the UI banners or snackbars to not confuse the user. But this is certainly something that should always be available and easy to find

Comment (ImmanuelSegol) [reply src/views/swap/steps.js]: @michael1011 @peartobear I followed this design https://app.zeplin.io/project/5bf801004f961c3f22c73d24/screen/5c308b0808e761af2e35498f

@peartobear I think it should always be easily available to the user.

File: src/views/swap/steps.js (hunk 1 above)
Comment (ImmanuelSegol) [reply src/views/swap/steps.js]: Yeah, I noticed, thanks for the heads up tho.

**Comment by ImmanuelSegol** [comment]
> I get this error when trying to test the branch:
> image

FIxed it, for some reason I didn't have this error till now.

File: src/views/swap/steps.js (hunk 2 above)
Comment (michael1011) [reply src/views/swap/steps.js]: @ImmanuelSegol yeah I know but @peartobear shouldn't have removed it in the first place. Once you add this back in this PR can be merged.

File: src/views/swap/steps.js [hunk 3]
```diff
@@ -204,41 +204,37 @@ class StyledStepThree extends React.Component {
...
+              })}`}
+              download={'refund.json'}
+            >
+              Click here
+            </a>{' '}
+            if the download of &lsquo;refund.json&lsquo; <br /> didn&apos;t
+            start automatically.
+          </p>
+          <p className={classes.address}>
+            This refund file can be used to trsutlessly <br />
```
Comment (michael1011) [inline src/views/swap/steps.js]: Typo: 
```suggestion
            This refund file can be used to trustlessly <br />
```

File: src/views/swap/swap.js [hunk 4]
```diff
@@ -83,23 +86,34 @@ const Swap = ({
...
+                />
               )}
             />
             <StepsWizard.Control
               num={3}
-              render={() => (
+              render={props => (
                 <Controls
-                  text={swapStatus}
+                  text={'Ive downloaded the refund file'}
```
Comment (michael1011) [inline src/views/swap/swap.js]:
```suggestion
                  text={'I have downloaded the refund file'}
```

File: src/views/swap/swapReducer.js [hunk 5]
```diff
@@ -49,7 +53,10 @@ const reducer = (state = initalState, action) => {
         ...state,
         swapInfo: {
           ...state.swapInfo,
-          invoice: action.payload,
+          invoice: {
+            ...state.swapInfo,
+            ...action.payload,
```
Comment (michael1011) [inline src/views/swap/swapReducer.js]: Why not leave it the way it was? The invoice should just be `action.payload` shouldn't it?
